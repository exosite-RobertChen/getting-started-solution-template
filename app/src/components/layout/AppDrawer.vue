<template>
  <v-navigation-drawer
    app
    fixed
    width="200"
    :value="value"
    @input="$emit('input', $event)"
  >
    <v-row class="secondary" style="height: 100%">
      <v-col>
        <div class="text-center">
          <DynamicImage
            :image-sources="[
              'vendor/brandlogo-dark.png',
              'vendor/brandlogo.png'
            ]"
            image-class="my-4 mx-auto"
            image-alt="brandlogo"
            image-style="align-self: center; width: 90%"
          />
        </div>
        <v-list>
          <v-list-item
            v-for="(item, i) in drawerItems"
            :key="i"
            :to="item.route ? { name: item.route } : null"
            active-class="primary--text active-link"
            class="white-text darken-2 drawer__tile py-1"
            @click="item.click ? item.click() : null"
          >
            <v-list-item-action>
              <v-icon class="grey--text lighten-2">
                {{ item.icon }}
              </v-icon>
            </v-list-item-action>
            <v-list-item-content style="margin-left: -8px">
              <v-list-item-title
                class="drawer-item-title"
                v-text="item.title"
              />
            </v-list-item-content>
          </v-list-item>
        </v-list>
      </v-col>
      <div class="grey--text mt-auto drawer__company-footer text-center mb-2">
        <span> {{ branding.companyName }} &copy; 2019 </span>
      </div>
    </v-row>
  </v-navigation-drawer>
</template>

<script>
import DynamicImage from '@/components/shared/DynamicImage'

export default {
  name: 'AppDrawer',
  components: { DynamicImage },
  props: {
    value: {
      type: Boolean,
      default: false
    },
    branding: {
      type: Object,
      required: true
    }
  },
  data() {
    return {
      drawerItems: [
        {
          icon: 'devices',
          title: 'Devices',
          route: 'DevicesPage'
        },
        {
          icon: 'cloud',
          title: 'Apps',
          route: 'ApplicationsPage'
        },
        {
          icon: 'code',
          title: 'Api',
          click: () => {
            window.open(window.origin + '/docs')
          }
        }
      ]
    }
  }
}
</script>

<style lang="scss">
.v-navigation-drawer {
  /* width: 200px !important; */
  background-color: var(--v-secondary-base) !important;
}
.v-list__tile--active .v-list__tile__action:first-of-type .v-icon {
  color: var(--v-primary-base) !important;
}

.active-link {
  color: var(--v-primary-base) !important;
}

.drawer {
  &__tile {
    transition: all 0.5s ease !important;
    &:hover {
      background-color: var(--v-secondary-lighten1);
    }
  }
  &__company-footer {
    width: 100%;
  }
}
.theme--light.v-list-item:not(.v-list-item--active):not(.v-list-item--disabled)
  .drawer-item-title {
  color: white !important;
}
</style>
