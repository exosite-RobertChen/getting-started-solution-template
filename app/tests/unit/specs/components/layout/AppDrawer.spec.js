import Vue from 'vue'
import Vuetify from 'vuetify'
import merge from 'lodash.merge'
import { shallowMount } from '@vue/test-utils'
import AppDrawer from '@/components/layout/AppDrawer.vue'
import branding from '@/../vendor/branding.json'

const factory = overrides => {
  const localVue = Vue
  localVue.use(Vuetify)
  // to avoid 'component not registered' warning
  localVue.component(
    Vue.component('router-link', {
      render: () => `<div><slot /></div>`
    })
  )

  let defaults = {
    localVue,
    propsData: {
      value: true,
      branding
    },
    mocks: {
      $route: {
        name: 'DevicesPage'
      }
    }
  }
  merge(defaults, overrides)
  return shallowMount(AppDrawer, defaults)
}

describe('AppDrawer.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = factory()
  })
  afterEach(() => wrapper.destroy())

  test('can be shallowMounted', () => {
    expect(wrapper.contains(AppDrawer)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
