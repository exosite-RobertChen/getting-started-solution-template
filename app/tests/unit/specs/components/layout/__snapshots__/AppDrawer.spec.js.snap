// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDrawer.vue matches snapshot 1`] = `
<v-navigation-drawer-stub
  app="true"
  fixed="true"
  height="100vh"
  minivariantwidth="80"
  mobilebreakpoint="1264"
  src=""
  tag="nav"
  value="true"
  width="200"
>
  <v-row-stub
    class="secondary"
    tag="div"
  >
    <v-col-stub
      tag="div"
    >
      <div
        class="text-center"
      >
        <dynamicimage-stub
          imagealt="brandlogo"
          imageclass="my-4 mx-auto"
          imagesources="vendor/brandlogo-dark.png,vendor/brandlogo.png"
          imagestyle="align-self: center; width: 90%"
        />
      </div>
       
      <v-list-stub
        tag="div"
        tile="true"
      >
        <v-list-item-stub
          activeclass="primary--text active-link"
          class="white-text darken-2 drawer__tile py-1"
          tag="div"
          to="[object Object]"
        >
          <v-list-item-action-stub>
            <v-icon-stub
              class="grey--text lighten-2"
            >
              
              devices
            
            </v-icon-stub>
          </v-list-item-action-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              class="drawer-item-title"
            />
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub
          activeclass="primary--text active-link"
          class="white-text darken-2 drawer__tile py-1"
          tag="div"
          to="[object Object]"
        >
          <v-list-item-action-stub>
            <v-icon-stub
              class="grey--text lighten-2"
            >
              
              cloud
            
            </v-icon-stub>
          </v-list-item-action-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              class="drawer-item-title"
            />
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub
          activeclass="primary--text active-link"
          class="white-text darken-2 drawer__tile py-1"
          tag="div"
        >
          <v-list-item-action-stub>
            <v-icon-stub
              class="grey--text lighten-2"
            >
              
              code
            
            </v-icon-stub>
          </v-list-item-action-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub
              class="drawer-item-title"
            />
          </v-list-item-content-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-col-stub>
     
    <div
      class="grey--text mt-auto drawer__company-footer text-center mb-2"
    >
      <span>
         Exosite Â© 2019 
      </span>
    </div>
  </v-row-stub>
</v-navigation-drawer-stub>
`;
